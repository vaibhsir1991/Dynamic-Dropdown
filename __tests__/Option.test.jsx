import React from 'react';
import renderer from 'react-test-renderer';
import Option from '../src/Components/Option';

import './helpers/setup-enzyme';
import './helpers/setup-browser-env';
import { renderOption } from './renderers';

describe('Test Option component', () => {
    test('Renders value element', () => {
        const tree = renderer.create((
            <Option
                cls={(key) => key}
                name="Foo"
                value="foo"
                type="button"
                highlighted={false}
                selected={false}
                renderOption={renderOption}
                optionProps={{
                    tabIndex: 0,
                    onMouseDown: () => {},
                }}
            />
        )).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Option can be selected', () => {
        const tree = renderer.create((
            <Option
                cls={(key) => key}
                name="Foo"
                value="foo"
                type="button"
                highlighted={false}
                selected={true}
                renderOption={renderOption}
                optionProps={{
                    tabIndex: 0,
                    onMouseDown: () => {},
                }}
            />
        )).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Option can be highlighted', () => {
        const tree = renderer.create((
            <Option
                cls={(key) => key}
                name="Foo"
                value="foo"
                type="button"
                highlighted={true}
                selected={false}
                renderOption={renderOption}
                optionProps={{
                    tabIndex: 0,
                    onMouseDown: () => {},
                }}
            />
        )).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Can render custom option', () => {
        const tree = renderer.create((
            <Option
                cls={(key) => key}
                name="Foo"
                value="foo"
                type="button"
                highlighted={true}
                selected={false}
                optionProps={{
                    tabIndex: 0,
                    onMouseDown: () => {},
                }}
                renderOption={() => <x-custom-element />}
            />
        )).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
