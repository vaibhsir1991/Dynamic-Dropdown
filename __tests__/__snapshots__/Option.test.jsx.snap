// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Option component Can render custom option 1`] = `
<li
  className="row"
  data-index={null}
  data-value="foo"
  role="menuitem"
>
  <x-custom-element />
</li>
`;

exports[`Test Option component Option can be highlighted 1`] = `
<li
  className="row"
  data-index={null}
  data-value="foo"
  role="menuitem"
>
  <button
    className="option is-highlighted"
    disabled={false}
    onMouseDown={[Function]}
    tabIndex={0}
  >
    Foo
  </button>
</li>
`;

exports[`Test Option component Option can be selected 1`] = `
<li
  className="row"
  data-index={null}
  data-value="foo"
  role="menuitem"
>
  <button
    className="option is-selected"
    disabled={false}
    onMouseDown={[Function]}
    tabIndex={0}
  >
    Foo
  </button>
</li>
`;

exports[`Test Option component Renders value element 1`] = `
<li
  className="row"
  data-index={null}
  data-value="foo"
  role="menuitem"
>
  <button
    className="option"
    disabled={false}
    onMouseDown={[Function]}
    tabIndex={0}
  >
    Foo
  </button>
</li>
`;
